comms/src/connection/peer_connection/worker.rs

```diff
@@ -185,6 +193,10 @@ impl PeerConnectionWorker {
                         );
                         let payload = self.create_payload(frames)?;
                         peer_conn.send(payload)?;
+                        self.with_connection_stats_mut(|stats| {
```

Comment:
Slight code smell. This object is modifying the internals of another object.

Consider moving getters and setters to ConnectionStats itself.

e.g. `connection_stats.incr_msg_sent()` would be more manageable. Then, changing anything on the interior of ConnectionStats would be opaque to ConnectionWorker. For example, right now the whole struct is locked when making any change, whereas you might want to change things to `AtomicUsize` for example.

Reply:
Makes sense, nothing much to comment, I think you're right :)

Reply:
As an experiment, I tried just using an Arc without a lock on the whole struct (because the memory is shared between the worker and PeerConnection) and making the internals thread safe, however mutable references and Arcs don't mix - so seems there's no getting away from locking the struct  as the current architecture stands - will incorporant the the getters and setters  though